pkg/foundation/grpcutil/websocket.go

```diff
@@ -233,13 +248,18 @@ func (p *webSocketProxy) startWebSocketWrite(ctx context.Context, messages chan
 				// readFromHTTPResponse closed the channel.
 				err := conn.WriteMessage(websocket.CloseMessage, []byte{})
 				if err != nil {
-					p.logger.Warn(ctx).Err(err).Msg("[write] failed sending close message")
+					p.logger.Warn(ctx).Err(err).Msg("failed sending close message")
 				}
 				return
 			}
 
 			if err := conn.WriteMessage(websocket.TextMessage, message); err != nil {
-				p.logger.Warn(ctx).Err(err).Msg("[write] error writing websocket message")
+				// NB: if this connection has been closed by the client
+				// then that will cancel the request context, which in turn
+				// makes the request return `{"code":1,"message":"context canceled","details":[]}`.
+				// This proxy will try to write that, but will fail,
+				// because the connection has already been closed.
+				p.logger.Warn(ctx).Err(err).Msg("failed writing websocket message")
```

Comment:
Nice comment, how about we take that knowledge and use it in code to ignore that log message? 🙂
```suggestion
				if string(message) != `{"code":1,"message":"context canceled","details":[]}` {
					p.logger.Warn(ctx).Bytes("message", message).Err(err).Msg("failed writing websocket message")
				}
```

Reply:
Or another way to go around this would be to lower the level in that case:
```suggestion
				e := p.logger.Warn(ctx)
				if string(message) == `{"code":1,"message":"context canceled","details":[]}` {
					e = p.logger.Trace(ctx)
				}
				e.Bytes("message", message).Err(err).Msg("failed writing websocket message")
```

Reply:
Lowering the log level makes sense to me!